### PR TITLE
Muon nxsV2  period info

### LINKED ISSUE #31206: Update NexusV2 and PSI Muon Loaders
As part of #30939 the nexusv1 loader was updated to include extra period information. We want the v2 loader and PSI loader to have the same sample logs added. Consider #29715 when doing this issue

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**

The muon users will be switching to using the nxs V2 files soon, instead of the current nxs files. A key bit of functionality that was missing from the nxs V2 was the ability to read the period information. This PR adds the period information to the workspace, so it can be viewed via the period info widget in the muon GUI's.

**To test:**

This will be easier with 2 versions of mantid (master and the one you are testing). - if you do this load the nxs V2 into the testing version and the same run into master (just search for the run - it will find the nxs V1)

Load the muon data
Go to the grouping tab
Click the `periods` button
A pop up will appear and will be populated - if you have master open the information should match

Repeat the above for both single and multiple period data (I can send you some)

Fixes #31206. 

In the release notes

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
the failing test is unrelated

File: Framework/DataHandling/src/LoadMuonNexusV2NexusHelper.cpp [hunk 1]
```diff
@@ -12,6 +12,16 @@
 #include "MantidDataObjects/TableWorkspace.h"
 #include "MantidDataObjects/Workspace2D.h"
 
+namespace {
+template <typename type> std::string convertVectorToString(const std::vector<type> &vector) {
+  std::string stringOfVector = "";
+  for (auto value : vector) {
```
Comment (SilkeSchomann) [inline Framework/DataHandling/src/LoadMuonNexusV2NexusHelper.cpp:18]:
```suggestion
  for (const auto &value : vector) {
```

Comment (SilkeSchomann) [reply Framework/DataHandling/src/LoadMuonNexusV2NexusHelper.cpp:18]: Just a minor thing, apart from that it looks good.

**Review by SilkeSchomann — CHANGES_REQUESTED** [review]

**Review by SilkeSchomann — APPROVED** [review]
Ship it :-)

**Pull request merged**